### PR TITLE
#1 Api Profile Service

### PR DESCRIPTION
**Pull request by mehtabkayani**

### CODE REVIEW

**Comment by JoakimAnder** [comment]
server port needs to be changed

File: src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java [hunk 1]
```diff
@@ -2,13 +2,169 @@
...
+        if(user.getPhone() == null || !Utility.validatePhone(user.getPhone())){
+            throw new RuntimeException("Phone number must only contain numbers and be in the range of 10-13");
+        }
+
+        oldUser.setPhone(user.getPhone());
+
+
+
+       profileRepo.save(oldUser);
+       UserViewable userViewable = new UserViewable(oldUser);
```
Comment (JoakimAnder) [inline src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java:103]: oldUser = profileRepo.save(oldUser);
UserViewable userViewable = new UserViewable(oldUser);
return userViewable;

File: src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java [hunk 2]
```diff
@@ -2,13 +2,169 @@
...
+        Long checkTimeStamp = LocalDateTime.now().toEpochSecond(Utility.SERVER_OFFSET);
+        if(oldUser.getTokenLastValidDate() < checkTimeStamp){
+            throw new RuntimeException("Session timout");
+        }
+        if(oldUser.getPayStatus() != 99){
+            throw new RuntimeException("You are active in the game and can´t delete your account");
+        }
+         profileRepo.delete(oldUser);
+
+        return oldUser;
```
Comment (JoakimAnder) [inline src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java]: Should we return a user on delete?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
-spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.hibernate.ddl-auto=update
 spring.datasource.url=jdbc:mysql://${MYSQL_HOST:localhost}:3306/tips?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC
 spring.datasource.username=tipsuser
 spring.datasource.password=root
-server.port=8605
```
Comment (JoakimAnder) [inline src/main/resources/application.properties]: change server port

File: src/main/java/com/pappaspojkars/tips/profileservice/Utility.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Utility {
+
+    @Autowired
+   static ProfileRepo profileRepo;
```
Comment (JoakimAnder) [inline src/main/java/com/pappaspojkars/tips/profileservice/Utility.java]: why do we need ProfileRepo here?

File: src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java (hunk 1 above)
Comment (mehtabkayani) [reply src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java:103]: Fixed

File: src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java (hunk 2 above)
Comment (mehtabkayani) [reply src/main/java/com/pappaspojkars/tips/profileservice/ProfileController.java]: Fixed

File: src/main/java/com/pappaspojkars/tips/profileservice/Utility.java (hunk 4 above)
Comment (mehtabkayani) [reply src/main/java/com/pappaspojkars/tips/profileservice/Utility.java]: fixed

File: src/main/resources/application.properties (hunk 3 above)
Comment (mehtabkayani) [reply src/main/resources/application.properties]: fixed

**Pull request merged**